sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/aio/_data_lake_service_client_async.py

```diff
@@ -97,13 +97,14 @@ async def __aenter__(self):
 
     async def __aexit__(self, *args):
         await self._blob_service_client.close()
+        await super(DataLakeServiceClient, self).__aexit__(*args)
 
     async def close(self):
         # type: () -> None
         """ This method is to close the sockets opened by the client.
         It need not be used when using with a context manager.
         """
-        await self._blob_service_client.close()
+        await self.__aexit__()
```

Comment:
Just checking for my understanding -- is there any reason why someone would want to call `close()` and only close the BSC and not the DatalakeServiceClient as well? Or was this just improperly implemented previously